File: app/src/main/java/com/mobile/hinde/connection/Duration_Site.java
package com.mobile.hinde.connection;

import android.os.AsyncTask;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.BufferedInputStream;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.Calendar;

public class Duration_Site extends AsyncTask<String, Void, JSONObject> {

    private AsyncResponse delegate;

    public Duration_Site(AsyncResponse delegate){
        this.delegate = delegate;
    }

    @Override
    protected JSONObject doInBackground(String... target) {
        StringBuilder result = new StringBuilder();
        HttpURLConnection urlConnection = null;
        JSONObject json_Res = null;
        try {
            Calendar rightNow = Calendar.getInstance();
            String date = rightNow.get(Calendar.YEAR) + "-" + String.format("%02d",rightNow.get(Calendar.MONTH) + 1 ) + "-" + String.format("%02d",rightNow.get(Calendar.DAY_OF_MONTH) );
            URL url = new URL("https://space-time-12b72.firebaseapp.com/" + date + ".json");
            urlConnection = (HttpURLConnection) url.openConnection();
            InputStream in = new BufferedInputStream(urlConnection.getInputStream());
            BufferedReader reader = new BufferedReader(new InputStreamReader(in));
            String line;
            while((line = reader.readLine()) != null) {
                    result.append(line);
            }

            json_Res = new JSONObject(result.toString());

        }catch(MalformedURLException | JSONException mue){

        }catch(IOException ioe){

        } finally {
            if(urlConnection != null) {
                urlConnection.disconnect();
            }
        }

        return json_Res;

    }

    @Override
    protected void onPostExecute(JSONObject result) {
        delegate.processFinish(result);
    }
}

File: app/src/main/java/com/mobile/hinde/connection/DurationSite.kt
package com.mobile.hinde.connection

import android.os.AsyncTask
import org.json.JSONObject
import java.io.FileNotFoundException
import java.net.URL
import java.util.*

class DurationSite(private val delegate: AsyncResponse) : AsyncTask<String, Any, Any>(){

    override fun doInBackground(vararg params: String?): JSONObject {
        val rightNow = Calendar.getInstance()
        val date = "%d-%02d-%02d".format(rightNow.get(Calendar.YEAR), rightNow.get(Calendar.MONTH) + 1, rightNow.get(Calendar.DAY_OF_MONTH))
        return try {
            val apiResponse = URL("https://space-time-12b72.firebaseapp.com/%s.json".format(date)).readText()
            JSONObject(apiResponse)
        }catch (e: FileNotFoundException){
            JSONObject("{\"error\" : \"file not found \"}")
        }
    }

    override fun onPostExecute(result: Any) {
        delegate.processFinish(result)
    }
}